src/Compilers/Core/Portable/Operations/IDataFlowAnalyzer.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.FlowAnalysis
+{
+#if WORKSPACE
+    using TBasicBlock = BasicBlock;
+#else
```

Comment:
what is going on here?

Reply:
Compiler layer does dataflow analysis based on their internal builder types when building the publically exposed ControlFlowGraph. This dataflow analysis implementation is generic and works even for the public BasicBlock types (only difference being BasicBlockBuilder is mutable and BasicBlock is immutable). These defines enable interim sharing of code until we can make these APIs public.

---
In reply to: [229794952](https://github.com/dotnet/roslyn/pull/30777#discussion_r229794952) [](ancestors = 229794952)